More specifically 

```
================================== FAILURES ===================================
______________________ TestIntegration.test_s3_roundtrip ______________________
[gw0] win32 -- Python 3.6.3 C:\Miniconda3_64\envs\pandas\python.exe
self = <pandas.tests.io.test_s3.TestIntegration object at 0x000000E2853F0400>
    def test_s3_roundtrip(self):
        expected = pd.DataFrame({'A': [1, 2, 3], 'B': 'foo'})
    
        boto3 = pytest.importorskip('boto3')
        moto = pytest.importorskip('moto')
    
        with moto.mock_s3():
            conn = boto3.resource("s3", region_name="us-east-1")
            conn.create_bucket(Bucket="pandas-test")
    
>           expected.to_parquet('s3://pandas-test/test.parquet')
pandas\tests\io\test_s3.py:24: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
pandas\core\frame.py:1680: in to_parquet
    compression=compression, **kwargs)
pandas\io\parquet.py:227: in to_parquet
    return impl.write(df, path, compression=compression, **kwargs)
pandas\io\parquet.py:110: in write
    path, _, _ = get_filepath_or_buffer(path, mode='wb')
pandas\io\common.py:199: in get_filepath_or_buffer
    from pandas.io import s3
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
    """ s3 support for remote file interactivity """
    from pandas import compat
    try:
        import s3fs
        from botocore.exceptions import NoCredentialsError
    except:
>       raise ImportError("The s3fs library is required to handle s3 files")
E       ImportError: The s3fs library is required to handle s3 files
pandas\io\s3.py:7: ImportError
```

--- a/pandas/io/parquet.py
+++ b/pandas/io/parquet.py
@@ -194,16 +194,14 @@ class FastParquetImpl(BaseImpl):
         # thriftpy/protocol/compact.py:339:
         # DeprecationWarning: tostring() is deprecated.
         # Use tobytes() instead.
-        path, _, _ = get_filepath_or_buffer(path, mode='wb')
+        path, _, _ = get_filepath_or_buffer(path)
         with catch_warnings(record=True):
             self.api.write(path, df,
-                           compression=compression,
-                           open_with=lambda path, mode: path,
-                           **kwargs)
+                           compression=compression, **kwargs)
 
     def read(self, path, columns=None, **kwargs):
         path, _, _ = get_filepath_or_buffer(path)
-        parquet_file = self.api.ParquetFile(path, open_with=lambda path, mode: path)
+        parquet_file = self.api.ParquetFile(path)
         return parquet_file.to_pandas(columns=columns, **kwargs)
 
 

--- a/pandas/tests/io/test_parquet.py
+++ b/pandas/tests/io/test_parquet.py
@@ -203,6 +203,16 @@ def test_cross_engine_fp_pa(df_cross_compat, pa, fp):
         result = read_parquet(path, engine=pa, columns=['a', 'd'])
         tm.assert_frame_equal(result, df[['a', 'd']])
 
+def test_s3_roundtrip(df_compat, s3_resource, engine):
+    # GH #19134
+    if engine == 'pyarrow':
+        df_compat.to_parquet('s3://pandas-test/test.parquet',
+                             engine, compression=None)
+
+        expected = df_compat
+        actual = pd.read_parquet('s3://pandas-test/test.parquet', engine)
+
+        tm.assert_frame_equal(expected, actual)
 
 class Base(object):
 
@@ -487,15 +497,3 @@ class TestParquetFastParquet(Base):
             result = read_parquet(path, fp, filters=[('a', '==', 0)])
         assert len(result) == 1
 
-
-class TestIntegrationWithS3(Base):
-    def test_s3_roundtrip(self, df_compat, s3_resource, engine):
-        # GH #19134
-        if engine == 'pyarrow':
-            df_compat.to_parquet('s3://pandas-test/test.parquet',
-                                 engine, compression=None)
-
-            expected = df_compat
-            actual = pd.read_parquet('s3://pandas-test/test.parquet', engine)
-
-            tm.assert_frame_equal(expected, actual)

--- a/pandas/tests/io/test_parquet.py
+++ b/pandas/tests/io/test_parquet.py
@@ -488,18 +488,11 @@ class TestParquetFastParquet(Base):
         assert len(result) == 1
 
 class TestIntegrationWithS3(Base):
-    def test_s3_roundtrip(self):
-        expected = pd.DataFrame({'A': [1, 2, 3], 'B': 'foo'})
+    def test_s3_roundtrip(self, df_compat, s3_resource):
+        df_compat.to_parquet('s3://pandas-test/test.parquet')
 
-        boto3 = pytest.importorskip('boto3')
-        moto = pytest.importorskip('moto')
+        expected = df_compat
+        actual = pd.read_parquet('s3://pandas-test/test.parquet')
 
-        with moto.mock_s3():
-            conn = boto3.resource("s3", region_name="us-east-1")
-            conn.create_bucket(Bucket="pandas-test")
-
-            expected.to_parquet('s3://pandas-test/test.parquet')
-            actual = pd.read_parquet('s3://pandas-test/test.parquet')
-
-            tm.assert_frame_equal(actual, expected)
+        tm.assert_frame_equal(expected, actual)
 

